exporter/zipkinexporter/zipkin.go

```diff
@@ -267,10 +277,15 @@ const zipkinRemoteEndpointKey = "zipkin.remoteEndpoint.serviceName"
 
 func (ze *zipkinExporter) zipkinSpan(node *commonpb.Node, s *trace.SpanData) (zc zipkinmodel.SpanModel) {
 	localEndpointServiceName := ze.serviceNameOrDefault(node)
-	localEndpoint := zipkinEndpointFromNode(node, localEndpointServiceName, isLocalEndpoint)
+	localEndpoint := zipkinEndpointFromAttributes(&s.Attributes, localEndpointServiceName, isLocalEndpoint)
 
-	remoteServiceName := lookupAttribute(node, zipkinRemoteEndpointKey)
-	remoteEndpoint := zipkinEndpointFromNode(node, remoteServiceName, isRemoteEndpoint)
+	remoteServiceName := ""
+	if remoteServiceEntry, ok := s.Attributes[zipkinRemoteEndpointKey]; ok {
+		if remoteServiceName, ok = remoteServiceEntry.(string); ok {
+			delete(s.Attributes, zipkinRemoteEndpointKey)
```

Comment:
Same here. Exporters are not allowed to modify data.
Easiest and quickest solution is to clone the data when entering this function. It will slow down things though.
For the future we could add declarative support for data modification to exporters just like we did for processors. It would allow exporters to modify the data freely, yet avoid cloning if there is only one exporter (this can be only found out by whole pipeline config analysis and exporters cannot figure this out on their own).